Mobile-configuration.md

```diff
@@ -0,0 +1,67 @@
+# Standarize mobile configuration format
+
+## Introduction
+
+Introduce standard for system interoperability, by providing schemas that will be used to communicate between mobile sdk's and core services.
+
+## Problem Description
+
+Mobile configuration format is not standarized. It contains various fields that have no meaning for the mobile users.
+Configuration also exposes some internal fields. 
+
+```json
+{
+ "config": {
+          "headers": {},
+          "mysql_db": "unifiedpush",
+          "mysql_password": "unifiedpush",
+          "mysql_username": "unifiedpush",
+          "name": "unified-push-server",
+          "type": "unified-push-server",
+          "uri": "http://ups-myproject.192.168.37.1.nip.io"
+        },
+        "name": "unified-push-server"
+}
+```
+
+Most important parameters often have different names. For example some services contain `url` property and some `uri`.
+Service specific metadata prevents us from strict json parsing.
+
+## Expectations
+
+- Provide mobile configuration that has only esential fields for mobile users
+- Standarize format so users to not break functionalities once users will start using configuration
+ 
+## Proposed Solution
+
+Provide standard for mobile configuration that will be extendable and have backwards compatibility.
+
+### Mobile cli specifics
+
+Mobile-cli currently print configuration out to the std. 
+Mobile-cli should have option to create `mobile-services.json` file in specified location.
+
+### Json format specific
+
+#### Top level object
+
+```json
+{
+ "version" : "number",
+ "serverUrl" : "string",
+ "namespace": "string",
+ "services": { "type": "array", "items":"Service"}
+}
+```
+
+### Service object
+
+```json
+{
+  "id": "string",
+  "name": "string",
+  "type": "string",
```

Comment:
@wtrocki as we discussed on bluejeans a moment ago, it would be good to add a:
```
"bindings": ["id", "id"]
```

field to the service object too.

Reply:
+100 for that. In Json schema this will look like this:

```
"bindings": {
  "type": "array",
  "items": {
    "type":  { "$ref": "#/services/id" },
   }
}
```

Reply:
What is the purpose of adding the bindings? Not against it just curious. My imagined workflow is as follows:

1) developer provision sync 
2) get config
3) write code to do sync
Happiness
4) provision keycloak
5)integrate sync with keycloak
6) get new config
7 write code to login user and get token pass it to sync sdk
Happiness 

Wondering what the bindings does here? I couldn't imagine a situation where the developer is unaware that keycloak exists and that sync is now being protected.

Reply:
SDK networking layer needs to know. This dependency has significant impact on SDK architecture because implies that sync will also need keycloak sdk dependency etc. I think this area needs special proposal with involvement from the Services 1 team. 

I was talking about that with @aidenkeating and it seems that Core SDK needs to be aware if sync, keycloak are joined together. Same will happen when API will be there.

Reply:
I am not involved enough with the SDK to really comment here, but it seems really strange to me that the core sdk needs to know if keycloak and sync are joined. 
@aidenkeating any insight here would be good. 
I would have expected that a token would be retrieved from the keycloak sdk and simply passed to via the exposed API to anything that needed it.